src/sentry/migrations/0471_backfill_groupedmessage_unresolved_none_status.py

```diff
@@ -0,0 +1,83 @@
+# Generated by Django 2.2.28 on 2023-05-30 23:37
+
+from django.db import connection, migrations
+from psycopg2.extras import execute_values
+
+from sentry.models import GroupHistoryStatus, GroupStatus
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.types.group import GroupSubStatus
+from sentry.utils.query import RangeQuerySetWrapper
+
+BATCH_SIZE = 100
+
+UPDATE_QUERY = """
+    UPDATE sentry_groupedmessage
+    SET substatus = data.substatus
+    FROM (VALUES %s) as data (id, status, substatus)
+    WHERE sentry_groupedmessage.id = data.id and sentry_groupedmessage.status = data.status and sentry_groupedmessage.substatus is NULL
+"""
+
+
+def map_unresolved_none_substatus(apps, schema_editor):
+    Group = apps.get_model("sentry", "Group")
+    GroupHistory = apps.get_model("sentry", "GroupHistory")
+
+    cursor = connection.cursor()
+    batch = []
+
+    for group_id, status, substatus in RangeQuerySetWrapper(
+        Group.objects.all().values_list("id", "status", "substatus"),
+        result_value_getter=lambda item: item[0],
+    ):
+        if status is not GroupStatus.UNRESOLVED and substatus is not None:
+            continue
+
+        try:
+            most_recent_history = GroupHistory.objects.filter(
+                group_id=group_id,
+            ).latest("date_added")
+        except GroupHistory.DoesNotExist:
+            continue
```

Comment:
Having this query here might end up being pretty slow since we'll make it 200mm times. There's no index on `group, date_added` either but probably nbd since there's typically a low number of rows per group.

I'm fine with just giving it a shot and seeing if it runs fast enough though.

Reply:
Yeah, luckily there's not too many cases of groups where `status=Unresolved, substatus=None`. I sampled like 100k projects and the associated groups and there were only about 3k groups that met that criteria. So the majority of the backfill will just be skipping through the groups.

Reply:
Right, somehow I overlooked that initial `if`, that should be fine then